In [8]:
import requests
from requests.auth import HTTPBasicAuth
from keysof import chaves
from proxyconfig import proxies

url = "https://gateway.apilib.prefeitura.sp.gov.br/token"

data = {
    "grant_type": "client_credentials"
}

client_id = chaves.get('client_id_')
client_secret = chaves.get('client_secret_')


response = requests.post(
    url,
    data=data,
    auth=HTTPBasicAuth(client_id, client_secret),
    verify=False,
    proxies=proxies
)

acess_token = response.json()['access_token']

response

c:\Projetos\API-Prefeitura_SP\venv\Lib\site-packages\urllib3\connectionpool.py:1097: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.10.193.25'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(


<Response [200]>

In [9]:
endpoint = "https://gateway.apilib.prefeitura.sp.gov.br/sf/sof/v4/contasReceita"

params = {
    'anoExercicio': 2025
}

headers = {
    "accept": "application/json",
    "Authorization": f"Bearer {acess_token}"
}

response = requests.get(endpoint, headers=headers, params=params, proxies=proxies)

response.json()

{'metaDados': {'txtMensagemErro': '', 'qtdPaginas': 525, 'txtStatus': 'OK'},
 'lstReceita': [{'codReceita': '1000000000000000000000000',
   'txtDescricaoReceita': 'RECEITAS CORRENTES'},
  {'codReceita': '1100000000000000000000000',
   'txtDescricaoReceita': 'IMPOSTOS, TAXAS E CONTRIBUIÇÕES DE MELHORIA'},
  {'codReceita': '1110000000000000000000000',
   'txtDescricaoReceita': 'IMPOSTOS'},
  {'codReceita': '1111000000000000000000000',
   'txtDescricaoReceita': 'IMPOSTOS SOBRE O COMÉRCIO EXTERIOR'},
  {'codReceita': '1111010000000000000000000',
   'txtDescricaoReceita': 'IMPOSTO SOBRE A IMPORTAÇÃO'},
  {'codReceita': '1111011000000000000000000',
   'txtDescricaoReceita': 'IMPOSTO SOBRE A IMPORTAÇÃO'},
  {'codReceita': '1111020000000000000000000',
   'txtDescricaoReceita': 'IMPOSTO SOBRE A EXPORTAÇÃO'},
  {'codReceita': '1111021000000000000000000',
   'txtDescricaoReceita': 'IMPOSTO SOBRE A EXPORTAÇÃO'},
  {'codReceita': '1112000000000000000000000',
   'txtDescricaoReceita': 'IMPOSTOS SOBR

Exemplo da saída do comando acima:

{'metaDados': {'txtMensagemErro': '', 'qtdPaginas': 525, 'txtStatus': 'OK'},
 'lstReceita': [{'codReceita': '1000000000000000000000000',
   'txtDescricaoReceita': 'RECEITAS CORRENTES'},
  {'codReceita': '1100000000000000000000000',
   'txtDescricaoReceita': 'IMPOSTOS, TAXAS E CONTRIBUIÇÕES DE MELHORIA'},
   .
   .
   .

In [12]:
#O dataset de interesse está em:
response.json()['lstReceita']
#mas o resultado tem centenas de páginas, aqui o resultado é apenas a primeira. Cada página está limitada a 50 resultados

[{'codReceita': '1000000000000000000000000',
  'txtDescricaoReceita': 'RECEITAS CORRENTES'},
 {'codReceita': '1100000000000000000000000',
  'txtDescricaoReceita': 'IMPOSTOS, TAXAS E CONTRIBUIÇÕES DE MELHORIA'},
 {'codReceita': '1110000000000000000000000',
  'txtDescricaoReceita': 'IMPOSTOS'},
 {'codReceita': '1111000000000000000000000',
  'txtDescricaoReceita': 'IMPOSTOS SOBRE O COMÉRCIO EXTERIOR'},
 {'codReceita': '1111010000000000000000000',
  'txtDescricaoReceita': 'IMPOSTO SOBRE A IMPORTAÇÃO'},
 {'codReceita': '1111011000000000000000000',
  'txtDescricaoReceita': 'IMPOSTO SOBRE A IMPORTAÇÃO'},
 {'codReceita': '1111020000000000000000000',
  'txtDescricaoReceita': 'IMPOSTO SOBRE A EXPORTAÇÃO'},
 {'codReceita': '1111021000000000000000000',
  'txtDescricaoReceita': 'IMPOSTO SOBRE A EXPORTAÇÃO'},
 {'codReceita': '1112000000000000000000000',
  'txtDescricaoReceita': 'IMPOSTOS SOBRE O PATRIMÔNIO'},
 {'codReceita': '1112010000000000000000000',
  'txtDescricaoReceita': 'IMPOSTO SOBRE A PROP

Exemplo da saída do comando acima:

[{'codReceita': '1000000000000000000000000',
  'txtDescricaoReceita': 'RECEITAS CORRENTES'},
 {'codReceita': '1100000000000000000000000',
  'txtDescricaoReceita': 'IMPOSTOS, TAXAS E CONTRIBUIÇÕES DE MELHORIA'},
  .
  .
  .

In [ ]:
#A ideia aqui é iterar em cada página enquanto salva o resultado de cada página em um Dataframe pandas
import pandas as pd
#tabela = 

ModuleNotFoundError: No module named 'pandas'